Frends.PowerShell.RunScript/Frends.PowerShell.RunScript/PowerShell.cs

```diff
@@ -51,6 +55,64 @@ public static PowerShellResult RunScript(RunScriptInput input, [Browsable(false)
         });
     }
 
+    private static PowerShellResult ExecuteProcess(string scriptPath, PowerShellParameter[] parameters)
+    {
+        List<dynamic> results = new();
+        List<string> errors = new();
+
+        string parameterString = string.Empty;
+        if (parameters != null)
+        {
+            foreach (var param in parameters)
+            {
+                parameterString += $"-{param.Name} \"{param.Value}\" ";
+            }
+        }
+
+        using Process process = new();
+        process.StartInfo = new()
+        {
+            FileName = "powershell.exe",
+            Arguments = $"-NoProfile -ExecutionPolicy Bypass -File \"{scriptPath}\" {parameterString}",
+            UseShellExecute = false,
+            CreateNoWindow = true,
+            RedirectStandardError = true,
+            RedirectStandardOutput = true,
+        };
+
+        process.OutputDataReceived += (sender, args) =>
+        {
+            if (!string.IsNullOrWhiteSpace(args.Data))
+                results.Add(args.Data);
+        };
+
+        process.ErrorDataReceived += (sender, args) =>
```

Comment:
I think that we need to unwire event handlers for .NET to be able to clean up the `process` object. So those should be in the finally block.